pyomo/contrib/doe/doe.py

```diff
@@ -1302,133 +1216,1015 @@ def det_general(m):
+    def check_model_jac(self, jac=None):
+        if jac.shape != (self.n_experiment_outputs, self.n_parameters):
+            raise ValueError(
+                "Shape of Jacobian provided should be n experiment outputs by n parameters, or {} by {}, Jacobian provided has shape {} by {}".format(
+                    self.n_experiment_outputs,
+                    self.n_parameters,
+                    jac.shape[0],
+                    jac.shape[1],
+                )
+            )
+
+        self.logger.info("Jacobian provided matches expected dimensions from model.")
+
+    # Update the FIM for the specified model
+    def update_FIM_prior(self, model=None, FIM=None):
+        """
+        Updates the prior FIM on the model object. This may be useful when
+        running a loop and the user doesn't want to rebuild the model
+        because it is expensive to build/initialize.
+
+        Parameters
+        ----------
+        model: model where FIM prior is to be updated, Default: None, (self.model)
+        FIM: 2D np array to be the new FIM prior, Default: None
+        """
+        if model is None:
+            model = self.model
+
+        # Check FIM input
+        if FIM is None:
+            raise ValueError(
+                "FIM input for update_FIM_prior must be a 2D, square numpy array."
+            )
+
+        if not hasattr(model, "fim"):
+            raise RuntimeError(
+                "``fim`` is not defined on the model provided. Please build the model first."
+            )
+
+        self.check_model_FIM(model=model, FIM=FIM)
+
+        # Update FIM prior
+        for ind1, p1 in enumerate(model.parameter_names):
+            for ind2, p2 in enumerate(model.parameter_names):
+                model.prior_FIM[p1, p2].set_value(FIM[ind1, ind2])
+
+        self.logger.info("FIM prior has been updated.")
+
+    # ToDo: Add an update function for the parameter values? --> closed loop parameter estimation?
+    # Or leave this to the user?????
+    def update_unknown_parameter_values(self, model=None, param_vals=None):
+        raise NotImplementedError(
+            "Updating unknown parameter values not yet supported."
+        )
+
+    # Evaluates FIM and statistics for a full factorial space (same as run_grid_search)
+    def compute_FIM_full_factorial(self, design_ranges=None, method="sequential"):
+        """
+        Will run a simulation-based full factorial exploration of
+        the experimental input space (i.e., a ``grid search`` or
+        ``parameter sweep``) to understand how the FIM metrics
+        change as a function of the experimental design space.
+
+        Parameters
+        ----------
+        model: model to perform the full factorial exploration on
+        design_ranges: dict of lists, of the form {<var_name>: [start, stop, numsteps]}
+        method: string to specify which method should be used
+                options are ``kaug`` and ``sequential``
```

Comment:
The docstring doesn't match the implementation. `model` isn't in the argument list.